docs/developer/extending/apps/developming-with-tunnels.mdx

```diff
@@ -0,0 +1,82 @@
+---
+title: Developing apps with tunnels
+---
+
+Saleor apps can be developed fully locally or using Saleor deployed to the cloud (either Saleor Cloud or a self-hosted instance on the internet).
+
+This article describes developing apps using tunnelling.
+
+## How does it work
+
+Saleor must communicate with your app, but it can't reach your localhost. Tunnelling software creates a proxy,
+allowing access to a specific port from localhost to a publically accessible URL
+
+<!-- Consider diagram -->
+
+## Available tunnels
+
+- [Saleor App Tunnel](#saleor-app-tunnel) - official tunnel maintained by Saleor, built into Saleor CLI
+- [ngrok](https://ngrok.com/) - popular tunnelling software with free plans
+
+You can also look into [alternatives](https://github.com/anderspitman/awesome-tunneling)
+
+## Using tunnels
+
+Tunneling software is exposing an app running on localhost on a specific port. Before you start, make sure dev server is on.
+
+Typically, running Next.js app in development mode, `next dev` will open the app on `localhost:3000`
+
+### Saleor App Tunnel
+
+To use a tunnel, you must have installed [Saleor CLI](https://www.npmjs.com/package/@saleor/cli)
+
+:::note
+Run `saleor app tunnel -h` to check detailed usage
+:::
+
+To create a tunnel, run `saleor app tunnel`. After several seconds, CLI should print the URL with `.saleor.live` domain:
+
+```bash
+$ saleor app tunnel
+
+✔ Organization · YourShop
+✔ Select Environment · staging
+✔ Environment · staging
+─────────────────────────────────────────────────────────────────
+
+     Saleor App Name: Your app
+     Saleor App URL: https://your-app-shop.saleor.live
+
+     Saleor Dashboard: https://yourshop.eu.saleor.cloud/dashboard/
+     GraphQL Playground: https://yourshop.eu.saleor.cloud/graphql/
+
+─────────────────────────────────────────────────────────────────
+✔ Installing... 
+
+Tunnel is listening to your local machine on port: 3000
+
+Press CTRL-C to stop the tunnel
+Use it to install the app in the Dashboard
```

Comment:
Please add three ` to close code snippet.
```suggestion
Use it to install the app in the Dashboard
\`\`\`

```

Code snipped not closed. Don't know how to escape it properly inside code suggestion 🤔